.github/workflows/ci.yml

```diff
@@ -43,12 +48,11 @@ jobs:
         sudo ./.ci/install_debian.sh
 
     - name: Dependencies [MacOS]
-      if: matrix.os == 'macOS-latest'
+      if: matrix.os == 'macOS-10.14'
       run: |
+        brew cask install xquartz
         brew install ace asio eigen tinyxml swig qt5 gsl pkg-config jpeg tinyxml
-        # Understand how to export env variables in a os-specific way
-        export Qt5_DIR=/usr/local/opt/qt5/lib/cmake/Qt5
```

Comment:
Curiosity: is this no longer necessary?

Reply:
I have no idea, but I wanted to test it.

Reply:
Apparently it is not necessary anymore, as the test are successful even without setting it.

Reply:
Apparently it is still necessary, as without it the Qt5 GUI are skipped, see in the log: 
~~~
2019-11-07T17:11:17.9116700Z -- Qt5 and GTK2 not found/selected, skipping iCubSkinGui
2019-11-07T17:11:17.9117690Z -- Qt5 and GTK2 not found/selected, skipping frameGrabberGui2
2019-11-07T17:11:17.9118630Z -- Qt5 and GTK2 not found/selected, skipping skinManagerGui
~~~

Reply:
@diegoferigo any idea on how to set `export Qt5_DIR=/usr/local/opt/qt5/lib/cmake/Qt5` only for a macos runner without duplicating the build step?

Reply:
Yes, there's an [undocumented method](https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps) that is all but future proof:

```yaml
- name: Setting env var
  if: matrix.os == macOS-10.14
  run: |
    echo "::set-env name=Qt5_DIR::/path/to/dir"
    env
```

Reply:
Ah cool. For the time being I just set `Qt5_DIR` to `/usr/local/opt/qt5/lib/cmake/Qt5` also in  Linux, as `/usr/local/opt/qt5/lib/cmake/Qt5` is an empty directory in Linux and qt is found just by using the system paths, but this method may be useful.

Reply:
It seems that there are some docs about it, see : https://help.github.com/en/actions/automating-your-workflow-with-github-actions/development-tools-for-github-actions#set-an-environment-variable-set-env . Thanks @pattacini for the link.

Reply:
Cool, thanks for letting me know! Also the logger support is quite interesting